### PR TITLE
Add base profile example and mergeable profiles

### LINKED ISSUE #150: Decide whether and how to add default syscalls to the SeccompProfile CRD
Part of the idea of #117 was to perhaps ensure that a standard set of default syscalls were present in a profile so that a minimum viable pod could be launched in kubernetes on e.g. runc. This issue is to evaluate whether this is worthwhile and discuss how to implement it.

The first consideration is that it's possible that the syscalls used by runc could change over time, and in fact they haven't been fully documented (https://github.com/opencontainers/runc/issues/2097) so we would need to maintain our own list. Moreover, a container runtime other than runc might be used, and container managers on top of runc like docker/cri-o might also call other syscalls. Also, the list might be different depending on the architecture.

Another consideration is the pause container, which seems to use an additional set of syscalls on top of what is required for runc. Should those syscalls be included in the defaults?

Finally, the actions to take for a syscall aren't a blanket "allow" or "deny", there is also e.g. "log", "trace", "kill", etc. If a user creates a complex profile with different actions for different syscalls, should the default syscalls use the "allow" action or the "log" action? What happens if the user adds one of the default syscalls to their "log" rule, but the controller is automatically adding it to the "allow" rule?

One idea could be to simply maintain an example SeccompProfile that contains the minimum required syscalls instead of building it into the CRD/controller. Thoughts?

### PR DESCRIPTION
**Pull request by cmurphy**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

This PR has two commits, one to add a base `SeccompProfile` resource containing the minimum required syscalls to launch a container on runc in kubernetes, and another to add a `BaseProfileName` attribute to the `SeccompProfile` CRD to allow users to construct profiles whose syscalls take the union of two profiles.

#### Which issue(s) this PR fixes:

Fixes #150 

#### Does this PR introduce a user-facing change?

```release-note
Added a new example `SeccompProfile` to provide a starting point on which to build custom profiles, and an attribute `BaseProfileName` to the `SeccompProfile` kind to allow merging syscalls from two profiles.
```

### CODE REVIEW

File: examples/baseprofile.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+---
+apiVersion: security-profiles-operator.x-k8s.io/v1alpha1
+kind: SeccompProfile
+metadata:
+  name: base
```
Comment (saschagrunert) [inline examples/baseprofile.yaml:5]: Do you think we need some kind of description field for a profile, too? This way we could still name it `base`, but explain for which runtime (runc) and version it has been built and for which purpose.

File: installation-usage.md [hunk 2]
```diff
@@ -81,6 +81,32 @@ spec:
       image: nginx
 ```
 
+#### Base syscalls for a container runtime
+
+An example of the minimum required syscalls for a runtime such as runc to launch
```
Comment (saschagrunert) [inline installation-usage.md:86]: Let's include the runc version:
```suggestion
An example of the minimum required syscalls for a runtime such as runc v1.0.0-rc92 to launch
```

File: internal/pkg/controllers/profile/profile.go [hunk 3]
```diff
@@ -120,14 +121,76 @@ func (r *Reconciler) Reconcile(req reconcile.Request) (reconcile.Result, error)
...
+		for n := range names {
+			syscall.Names = append(syscall.Names, n)
+		}
+		finalSyscalls = append(finalSyscalls, &syscall)
+	}
+	return finalSyscalls
+}
+
+func (r *Reconciler) mergeBaseProfile(
+	ctx context.Context, sp *v1alpha1.SeccompProfile, l logr.Logger) (OutputProfile, error) {
```
Comment (saschagrunert) [inline internal/pkg/controllers/profile/profile.go:160]: Just a style nit:
```suggestion
	ctx context.Context, sp *v1alpha1.SeccompProfile, l logr.Logger,
) (OutputProfile, error) {
```

**Review by saschagrunert — APPROVED** [review]
Just some nits, otherwise LGTM :)

File: examples/baseprofile.yaml (hunk 1 above)
Comment (cmurphy) [reply examples/baseprofile.yaml:5]: Maybe we could just include it in the name, combined with `targetWorkload`? e.g. `name: runc-v1.0.0-rc92` `targetWorkload: base`

File: installation-usage.md (hunk 2 above)
Comment (cmurphy) [reply installation-usage.md:86]: I didn't happen to test it on `v1.0.0-rc92`, have you? I've only tried `v1.0.0-rc10`

File: examples/baseprofile.yaml (hunk 1 above)
Comment (saschagrunert) [reply examples/baseprofile.yaml:5]: Yes this would also work from my perspective :+1:

File: installation-usage.md (hunk 2 above)
Comment (saschagrunert) [reply installation-usage.md:86]: No, I just assumed the latest version. I would not expect changes in the setup/syscalls, but who knows…

File: examples/baseprofile.yaml (hunk 1 above)
Comment (cmurphy) [reply examples/baseprofile.yaml]: done

File: installation-usage.md (hunk 2 above)
Comment (cmurphy) [reply installation-usage.md]: tested on rc92, added the version here and to the yaml

File: internal/pkg/controllers/profile/profile.go (hunk 3 above)
Comment (cmurphy) [reply internal/pkg/controllers/profile/profile.go]: done

**Review by cmurphy — COMMENTED** [review]
Tested on runc v1.0.0-rc92, also fixed the e2e test to actually use a security context :facepalm:

**Review by saschagrunert — APPROVED** [review]
/lgtm
Thank you!

**Pull request merged**